CMakeLists.txt

```diff
@@ -7,6 +7,7 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake-modules")
 # Compile Options
 option(WARNINGS_AS_ERRORS "Treat compiler warnings as errors" ON)
 option(BUILD_CURL_TRANSPORT "Build internal http transport implementation with CURL for HTTP Pipeline" OFF)
+option(BUILD_TESTING "Build test cases" OFF)
```

Comment:
Nit: BUILD_TESTS

Reply:
i was told by Charlie that most of C++ projects outside uses `BUILD_TESTING` . And that it's like almost idiomatic.

Reply:
Ok, I'll trust Charlie and google.  I did a quick search and see BUILD_TESTING all over the internet.  Lets go with BUILD_TESTING